app/app.go

```diff
@@ -1063,6 +1067,21 @@ func (app *EthermintApp) GetStoreKey(name string) storetypes.StoreKey {
 	return app.okeys[name]
 }
 
+// RegisterPendingTxListener is used by json-rpc server to listen to pending transactions in CheckTx.
+func (app *EthermintApp) RegisterPendingTxListener(listener PendingTxListener) {
+	app.pendingTxListeners = append(app.pendingTxListeners, listener)
+}
+
+func (app *EthermintApp) CheckTx(req *abci.RequestCheckTx) (*abci.ResponseCheckTx, error) {
```

Comment:
One alternative is to trigger this in an ante handler decorator, so the listener don't need to parse the tx body again. but the decorator must be put at the last of the chain, so when it's executed, the ante handler must succed.